libraries/designsystem/src/main/kotlin/io/element/android/libraries/designsystem/text/UnitConverters.kt

```diff
@@ -52,3 +52,19 @@ fun Dp.toPx(): Float = with(LocalDensity.current) { toPx() }
  */
 @Composable
 fun Dp.roundToPx(): Int = with(LocalDensity.current) { roundToPx() }
+
+/**
+ * Return the maximum value between the receiver value and the value with fonScale applied.
+ */
+@Composable
+fun Dp.scaleMax(): Dp = with(LocalDensity.current) {
+    return this@scaleMax * fontScale.coerceAtMost(1f)
+}
+
+/**
+ * Return the minimum value between the receiver value and the value with fonScale applied.
+ */
+@Composable
+fun Dp.scaleMin(): Dp = with(LocalDensity.current) {
+    return this@scaleMin * fontScale.coerceAtLeast(1f)
```

Comment:
I might be missing something, but these feel backwards. If the font scale is 1.5, then `scaleMin` will return `1.5x` but `scaleMax` will return `1.0x`? Feels very weird to have min > max :thinking:

Reply:
I also find the API a bit confusing but I think it is a confusing concept.

How about combining the `scale()` function so that it covers the most common use case with default parameters?

```kotlin
/**
 * Apply the system font scale to a density pixel value.
 * 
 * Specify min or max to limit the scaling applied. By default, the 
 * applied scale is limited to a minimum of 1.
 */
fun Dp.scale(
    min: Float = 1f
    max: Float = Float.MAX_VALUE
): Dp = with(LocalDensity.current) {
    this@scaleMin * fontScale.coerceAtLeast(min).coerceAtMost(max)
}

Reply:
Thanks for the suggestion. I can definitely improve this. The "problem" with fun `Dp.scale` - IIUC - is that you can use it like this:

```
16.dp.scale()
```

to ensure that the final size will be minimum `16.dp`, or like this:

```
16.dp.scale(0f, 1f)
```

to ensure that the final size will be maximum `16.dp`, and I do not like this oddity :).

I will try to find something better.

Reply:
[5ace169](https://github.com/vector-im/element-x-android/pull/986/commits/5ace169440daf61320a14e7eb3ae7ffa92c34e38), let me know what you think!

Reply:
I am happy with this although (for me) it is a bit verbose for a utility which will likely be littered all over the codebase.

At the risk of bike shedding, one more suggestion:

```kotlin
enum class DpScaleMode {
    Exact, UpOnly, DownOnly
}

fun Dp.scale(
    mode: DpScaleMode = DpScaleMode.UpOnly,
): Dp ...
```

At the end of the day I think it is a style thing, so am happy to go with your suggestion!

Reply:
(I do not think we will use those 2 extensions a lot, only on the most important part of the app I guess)